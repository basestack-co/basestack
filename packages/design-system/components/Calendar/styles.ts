import styled from "styled-components";
import { rem } from "polished";

export const Container = styled.div`
  .react-calendar,
  .react-calendar *,
  .react-calendar *:before,
  .react-calendar *:after {
    box-sizing: border-box;
  }

  .react-calendar {
    padding: ${({ theme }) => theme.spacing.s2};
    width: 350px;
    max-width: 100%;
    background: ${({ theme }) => theme.calendar.backgroundColor};
    box-shadow: ${({ theme }) => theme.shadow.elevation4};
    border-radius: ${rem("8px")};
    overflow: hidden;
    border: none;
    font-family: ${({ theme }) => theme.typography.roboto};
    line-height: 1.2;

    abbr[title] {
      text-decoration: none;
    }

    button {
      margin: 0;
      border: 0;
      outline: none;

      &:enabled:hover {
        cursor: pointer;
      }
    }

    /* Double View Styles */
    &--doubleView {
      width: 700px;

      .react-calendar__viewContainer {
        display: flex;
        margin: -0.5em;

        > * {
          width: 50%;
          margin: 0.5em;
        }
      }
    }

    /* Navigation Styles */
    &__navigation {
      height: ${rem("40px")};
      display: flex;
      margin-bottom: ${({ theme }) => theme.spacing.s1};

      button {
        min-width: ${rem("40px")};
        height: ${rem("40px")};
        font-size: ${rem("14px")};
        background: none;

        &:enabled:hover,
        &:enabled:focus {
          background-color: ${({ theme }) =>
            theme.calendar.navigation.button.hover.backgroundColor};
        }

        &[disabled] {
          background-color: ${({ theme }) =>
            theme.calendar.navigation.button.disabled.backgroundColor};
        }
      }

      &__label {
        color: ${({ theme }) => theme.calendar.day.color};
        align-items: center;
        justify-content: center;
        padding: ${({ theme }) => theme.spacing.s1};
        border-radius: ${rem("4px")};
      }

      &__arrow {
        border-radius: ${rem("4px")};
        display: flex;
        align-items: center;
        justify-content: center;

        &:disabled span {
          color: ${({ theme }) =>
            theme.calendar.navigation.button.disabled.color};
        }
      }
    }

    /* Body Styles */
    &__month-view__weekdays {
      text-align: center;
      font-weight: 500;
      font-size: ${rem("14px")};
      margin-bottom: ${({ theme }) => theme.spacing.s1};

      &__weekday {
        color: ${({ theme }) => theme.calendar.day.color};
        padding: ${({ theme }) => theme.spacing.s2}
          ${({ theme }) => theme.spacing.s1};
      }
    }

    /* Start TODO find out where these styles are in the calendar */
    &__month-view__weekNumbers {
      font-weight: 500;

      .react-calendar__tile {
        display: flex;
        align-items: center;
        justify-content: center;
        font-size: 0.75em;
        padding: calc(0.75em / 0.75) calc(0.5em / 0.75);
      }
    }
    /* End TODO find out where these styles are in the calendar */

    &__month-view__days {
      &__day {
        color: ${({ theme }) => theme.calendar.day.color};
        font-size: ${rem("14px")};

        &:disabled {
          color: ${({ theme }) => theme.calendar.day.disabled.color};
        }

        &--weekend {
          color: ${({ theme }) => theme.calendar.day.weekend.color};
        }

        &--neighboringMonth {
          color: ${({ theme }) => theme.calendar.day.neighboringMonth.color};
        }
      }
    }

    &__tile {
      color: ${({ theme }) => theme.calendar.title.color};
      max-width: 100%;
      text-align: center;
      padding: ${({ theme }) => theme.spacing.s3}
        ${({ theme }) => theme.spacing.s2};
      background: none;
      border-radius: ${rem("4px")};

      &:disabled {
        color: ${({ theme }) => theme.calendar.title.disabled.color};
        background-color: ${({ theme }) =>
          theme.calendar.title.disabled.backgroundColor};
      }

      &:enabled:hover,
      &:enabled:focus {
        background-color: ${({ theme }) =>
          theme.calendar.title.hover.backgroundColor};
      }

      &--now {
        background: ${({ theme }) => theme.calendar.now.backgroundColor};

        &:enabled:hover,
        &:enabled:focus {
          background: ${({ theme }) =>
            theme.calendar.now.hover.backgroundColor};
        }
      }

      &--hasActive {
        background: ${({ theme }) => theme.calendar.hasActive.backgroundColor};
        color: ${({ theme }) => theme.calendar.hasActive.color};

        &:enabled:hover,
        &:enabled:focus {
          background: ${({ theme }) =>
            theme.calendar.hasActive.hover.backgroundColor};
        }
      }

      &--active {
        background: ${({ theme }) => theme.calendar.active.backgroundColor};
        color: ${({ theme }) => theme.calendar.active.color};

        &:enabled:hover,
        &:enabled:focus {
          background: ${({ theme }) =>
            theme.calendar.active.hover.backgroundColor};
        }
      }
    }

    &--selectRange {
      .react-calendar__tile--hover {
        background-color: ${({ theme }) =>
          theme.calendar.selectRange.backgroundColor};
      }
    }

    &__year-view,
    &__decade-view,
    &__century-view {
      .react-calendar__tile {
        padding: ${({ theme }) => theme.spacing.s5}
          ${({ theme }) => theme.spacing.s2};
      }
    }
  }
`;
