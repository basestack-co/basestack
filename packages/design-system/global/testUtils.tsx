import { type RenderOptions, render } from "@testing-library/react";
import type React from "react";
// Styles
import { ThemeProvider } from "styled-components";
import theme from "../theme/lightTheme";

interface Props {
  children: React.ReactNode;
}

const WithThemeProvider = ({ children }: Props) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

const renderWithTheme = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
) => render(ui, { wrapper: WithThemeProvider, ...options });

export { renderWithTheme };
