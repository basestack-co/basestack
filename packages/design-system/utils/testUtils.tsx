import defaultTheme from "@basestack/design-system/theme/lightTheme";
import isPropValid from "@emotion/is-prop-valid";
import { RenderResult, render } from "@testing-library/react";
import React from "react";
import { StyleSheetManager, ThemeProvider } from "styled-components";

const Provider = ({
  children,
  theme = defaultTheme,
}: {
  children: React.ReactNode;
  theme: any;
}) => (
  <ThemeProvider theme={theme}>
    <StyleSheetManager shouldForwardProp={isPropValid}>
      {children}
    </StyleSheetManager>
  </ThemeProvider>
);

export const renderWithTheme = (
  ui: React.ReactElement,
  options?: any,
): RenderResult => render(ui, { wrapper: Provider, ...options });
