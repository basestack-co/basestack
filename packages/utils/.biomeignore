# Node dependencies
node_modules

# Next.js build output
.next
out

# Environment files
.env
.env.*
!.env.example

# Logs
*.log
npm-debug.log*
yarn-debug.log*
yarn-error.log*
pnpm-debug.log*

# OS / editor files
.DS_Store
Thumbs.db

# Package manager cache
.pnp
.pnp.js
.pnp.cjs

# Test coverage
coverage

# Build artifacts
dist
build

# Temporary files
.tmp
temp