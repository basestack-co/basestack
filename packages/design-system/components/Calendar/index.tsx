import { forwardRef, memo } from "react";
// Calendar
import Calendar, { type CalendarProps } from "react-calendar";
import Icon from "../Icon";
// Components
import { Container } from "./styles";

const CalendarComp = forwardRef<HTMLDivElement, CalendarProps>((props, ref) => {
  return (
    <Container ref={ref}>
      <Calendar
        {...props}
        prevLabel={<Icon icon="chevron_left" size="small" />}
        prev2Label={<Icon icon="keyboard_double_arrow_left" size="small" />}
        nextLabel={<Icon icon="chevron_right" size="small" />}
        next2Label={<Icon icon="keyboard_double_arrow_right" size="small" />}
      />
    </Container>
  );
});

CalendarComp.displayName = "CalendarComp";

export default memo(CalendarComp);
