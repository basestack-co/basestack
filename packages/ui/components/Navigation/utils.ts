// Types
import { Translate } from "next-translate";
import { NextRouter } from "next/router";
// Utils
import { config } from "@basestack/utils";
import { signOut } from "next-auth/react";

export const getInternalLinks = (t: Translate, formId: string) => {
  return [
    {
      icon: "flag",
      activeText: "Submissions",
      to: `/form/${formId}/submissions`,
      text: t("internal.submissions"),
    },
    {
      icon: "flag",
      activeText: "Setup",
      to: `/form/${formId}/setup`,
      text: t("internal.setup"),
    },
    {
      icon: "settings",
      activeText: "Settings",
      to: `/form/${formId}/settings/general`,
      text: t("internal.settings"),
    },
  ];
};

export const getExternalLinks = (t: Translate) => {
  return [
    {
      icon: "description",
      activeText: "Documentation",
      to: config.urls.docs.flags.base,
      isExternal: true,
      text: t("external.docs"),
    },
    {
      icon: "link",
      activeText: "Github",
      to: config.urls.repo,
      isExternal: true,
      text: t("external.github"),
    },
  ];
};

export const getAvatarDropdownList = (
  t: Translate,
  router: NextRouter,
  onCreateForm: () => void,
) => {
  return [
    {
      icon: "add_circle",
      text: t("create.form"),
      onClick: onCreateForm,
      separator: true,
    },
    {
      icon: "settings",
      text: t("dropdown.settings"),
      onClick: () =>
        router.push({
          pathname: "/user/profile/settings",
        }),
    },
    {
      icon: "logout",
      text: t("dropdown.logout"),
      onClick: signOut,
    },
  ];
};
