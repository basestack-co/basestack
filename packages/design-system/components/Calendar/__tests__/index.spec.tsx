import { cleanup } from "@testing-library/react";
import { renderWithTheme } from "../../../utils/testUtils";
import Calendar from "..";

jest.useFakeTimers();

describe("Calendar tests", () => {
  afterEach(cleanup);

  test("render Calendar correctly", () => {
    const { asFragment } = renderWithTheme(<Calendar />);
    expect(asFragment()).toMatchSnapshot();
  });
});
