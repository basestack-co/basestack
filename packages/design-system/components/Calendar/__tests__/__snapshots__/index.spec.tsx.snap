// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar tests render Calendar correctly 1`] = `
<DocumentFragment>
  .c0 .react-calendar,
.c0 .react-calendar *,
.c0 .react-calendar *:before,
.c0 .react-calendar *:after {
  box-sizing: border-box;
}

.c0 .react-calendar {
  padding: 0.5rem;
  width: 350px;
  max-width: 100%;
  background: #FFFFFF;
  box-shadow: rgba(0, 0, 0, 0.1) 0px 10px 15px -3px,rgba(0, 0, 0, 0.05) 0px 4px 6px -2px;
  border-radius: 0.5rem;
  overflow: hidden;
  border: none;
  font-family: 'Roboto',sans-serif;
  line-height: 1.2;
}

.c0 .react-calendar abbr[title] {
  text-decoration: none;
}

.c0 .react-calendar button {
  margin: 0;
  border: 0;
  outline: none;
}

.c0 .react-calendar button:enabled:hover {
  cursor: pointer;
}

.c0 .react-calendar--doubleView {
  width: 700px;
}

.c0 .react-calendar--doubleView .react-calendar__viewContainer {
  display: flex;
  margin: -0.5em;
}

.c0 .react-calendar--doubleView .react-calendar__viewContainer >* {
  width: 50%;
  margin: 0.5em;
}

.c0 .react-calendar__navigation {
  height: 2.5rem;
  display: flex;
  margin-bottom: 0.25rem;
}

.c0 .react-calendar__navigation button {
  min-width: 2.5rem;
  height: 2.5rem;
  font-size: 0.875rem;
  background: none;
}

.c0 .react-calendar__navigation button:enabled:hover,
.c0 .react-calendar__navigation button:enabled:focus {
  background-color: #EEEEEE;
}

.c0 .react-calendar__navigation button[disabled] {
  background-color: #FFFFFF;
}

.c0 .react-calendar__navigation__label {
  color: #000000;
  align-items: center;
  justify-content: center;
  padding: 0.25rem;
  border-radius: 0.25rem;
}

.c0 .react-calendar__navigation__arrow {
  border-radius: 0.25rem;
  display: flex;
  align-items: center;
  justify-content: center;
}

.c0 .react-calendar__navigation__arrow:disabled span {
  color: #AFAFAF;
}

.c0 .react-calendar__month-view__weekdays {
  text-align: center;
  font-weight: 500;
  font-size: 0.875rem;
  margin-bottom: 0.25rem;
}

.c0 .react-calendar__month-view__weekdays__weekday {
  color: #000000;
  padding: 0.5rem 0.25rem;
}

.c0 .react-calendar__month-view__weekNumbers {
  font-weight: 500;
}

.c0 .react-calendar__month-view__weekNumbers .react-calendar__tile {
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 0.75em;
  padding: calc(0.75em / 0.75) calc(0.5em / 0.75);
}

.c0 .react-calendar__month-view__days__day {
  color: #000000;
  font-size: 0.875rem;
}

.c0 .react-calendar__month-view__days__day:disabled {
  color: #AFAFAF;
}

.c0 .react-calendar__month-view__days__day--weekend {
  color: #000000;
}

.c0 .react-calendar__month-view__days__day--neighboringMonth {
  color: #AFAFAF;
}

.c0 .react-calendar__tile {
  color: #000000;
  max-width: 100%;
  text-align: center;
  padding: 0.75rem 0.5rem;
  background: none;
  border-radius: 0.25rem;
}

.c0 .react-calendar__tile:disabled {
  color: #AFAFAF;
  background-color: #FFFFFF;
}

.c0 .react-calendar__tile:enabled:hover,
.c0 .react-calendar__tile:enabled:focus {
  background-color: #EEEEEE;
}

.c0 .react-calendar__tile--now {
  background: #EEEEEE;
}

.c0 .react-calendar__tile--now:enabled:hover,
.c0 .react-calendar__tile--now:enabled:focus {
  background: #E2E2E2;
}

.c0 .react-calendar__tile--hasActive {
  background: #276EF1;
  color: #FFFFFF;
}

.c0 .react-calendar__tile--hasActive:enabled:hover,
.c0 .react-calendar__tile--hasActive:enabled:focus {
  background: #276EF1;
}

.c0 .react-calendar__tile--active {
  background: #276EF1;
  color: #FFFFFF;
}

.c0 .react-calendar__tile--active:enabled:hover,
.c0 .react-calendar__tile--active:enabled:focus {
  background: #276EF1;
}

.c0 .react-calendar__tile--range,
.c0 .react-calendar__tile--hover {
  border-radius: 0;
}

.c0 .react-calendar__tile--rangeStart:not(.c0 .react-calendar__tile--rangeBothEnds),
.c0 .react-calendar__tile--hoverStart:not(.c0 .react-calendar__tile--rangeBothEnds) {
  border-top-left-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.c0 .react-calendar__tile--rangeEnd:not(.c0 .react-calendar__tile--rangeBothEnds),
.c0 .react-calendar__tile--hoverEnd:not(.c0 .react-calendar__tile--rangeBothEnds) {
  border-top-right-radius: 0.25rem;
  border-bottom-right-radius: 0.25rem;
}

.c0 .react-calendar__tile--rangeBothEnds:not(.c0 .react-calendar__tile--hoverStart):not(.c0 .react-calendar__tile--hoverEnd) {
  border-radius: 0.25rem;
}

.c0 .react-calendar--selectRange .react-calendar__tile--hover,
.c0 .react-calendar--selectRange .react-calendar__tile:enabled:hover {
  background-color: #276EF1;
  color: #FFFFFF;
}

.c0 .react-calendar__year-view .react-calendar__tile,
.c0 .react-calendar__decade-view .react-calendar__tile,
.c0 .react-calendar__century-view .react-calendar__tile {
  padding: 1.25rem 0.5rem;
}

.c1 {
  height: 1.125rem;
  width: 1.125rem;
  flex-shrink: 0;
}

.c2 {
  color: #000000;
  font-size: 1.125rem;
}

<div
    class="c0"
  >
    <div
      class="react-calendar"
    >
      <div
        class="react-calendar__navigation"
      >
        <button
          aria-label=""
          class="react-calendar__navigation__arrow react-calendar__navigation__prev2-button"
          type="button"
        >
          <div
            class="c1"
            data-testid="icon-container"
            size="18"
          >
            <span
              class="c2 material-symbols-sharp"
              data-testid="icon"
              size="18"
            >
              keyboard_double_arrow_left
            </span>
          </div>
        </button>
        <button
          aria-label=""
          class="react-calendar__navigation__arrow react-calendar__navigation__prev-button"
          type="button"
        >
          <div
            class="c1"
            data-testid="icon-container"
            size="18"
          >
            <span
              class="c2 material-symbols-sharp"
              data-testid="icon"
              size="18"
            >
              chevron_left
            </span>
          </div>
        </button>
        <button
          aria-label=""
          class="react-calendar__navigation__label"
          style="flex-grow: 1;"
          type="button"
        >
          <span
            class="react-calendar__navigation__label__labelText react-calendar__navigation__label__labelText--from"
          >
            December 2023
          </span>
        </button>
        <button
          aria-label=""
          class="react-calendar__navigation__arrow react-calendar__navigation__next-button"
          type="button"
        >
          <div
            class="c1"
            data-testid="icon-container"
            size="18"
          >
            <span
              class="c2 material-symbols-sharp"
              data-testid="icon"
              size="18"
            >
              chevron_right
            </span>
          </div>
        </button>
        <button
          aria-label=""
          class="react-calendar__navigation__arrow react-calendar__navigation__next2-button"
          type="button"
        >
          <div
            class="c1"
            data-testid="icon-container"
            size="18"
          >
            <span
              class="c2 material-symbols-sharp"
              data-testid="icon"
              size="18"
            >
              keyboard_double_arrow_right
            </span>
          </div>
        </button>
      </div>
      <div
        class="react-calendar__viewContainer"
      >
        <div
          class="react-calendar__month-view"
        >
          <div
            style="display: flex; align-items: flex-end;"
          >
            <div
              style="flex-grow: 1; width: 100%;"
            >
              <div
                class="react-calendar__month-view__weekdays"
                style="display: flex; flex-wrap: nowrap;"
              >
                <div
                  class="react-calendar__month-view__weekdays__weekday"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                >
                  <abbr
                    aria-label="Monday"
                    title="Monday"
                  >
                    Mon
                  </abbr>
                </div>
                <div
                  class="react-calendar__month-view__weekdays__weekday"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                >
                  <abbr
                    aria-label="Tuesday"
                    title="Tuesday"
                  >
                    Tue
                  </abbr>
                </div>
                <div
                  class="react-calendar__month-view__weekdays__weekday"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                >
                  <abbr
                    aria-label="Wednesday"
                    title="Wednesday"
                  >
                    Wed
                  </abbr>
                </div>
                <div
                  class="react-calendar__month-view__weekdays__weekday"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                >
                  <abbr
                    aria-label="Thursday"
                    title="Thursday"
                  >
                    Thu
                  </abbr>
                </div>
                <div
                  class="react-calendar__month-view__weekdays__weekday react-calendar__month-view__weekdays__weekday--current"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                >
                  <abbr
                    aria-label="Friday"
                    title="Friday"
                  >
                    Fri
                  </abbr>
                </div>
                <div
                  class="react-calendar__month-view__weekdays__weekday react-calendar__month-view__weekdays__weekday--weekend"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                >
                  <abbr
                    aria-label="Saturday"
                    title="Saturday"
                  >
                    Sat
                  </abbr>
                </div>
                <div
                  class="react-calendar__month-view__weekdays__weekday react-calendar__month-view__weekdays__weekday--weekend"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                >
                  <abbr
                    aria-label="Sunday"
                    title="Sunday"
                  >
                    Sun
                  </abbr>
                </div>
              </div>
              <div
                class="react-calendar__month-view__days"
                style="display: flex; flex-wrap: wrap;"
              >
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 27, 2023"
                  >
                    27
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 28, 2023"
                  >
                    28
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 29, 2023"
                  >
                    29
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 30, 2023"
                  >
                    30
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 1, 2023"
                  >
                    1
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 2, 2023"
                  >
                    2
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 3, 2023"
                  >
                    3
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 4, 2023"
                  >
                    4
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 5, 2023"
                  >
                    5
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 6, 2023"
                  >
                    6
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 7, 2023"
                  >
                    7
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 8, 2023"
                  >
                    8
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 9, 2023"
                  >
                    9
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 10, 2023"
                  >
                    10
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 11, 2023"
                  >
                    11
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 12, 2023"
                  >
                    12
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 13, 2023"
                  >
                    13
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 14, 2023"
                  >
                    14
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 15, 2023"
                  >
                    15
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 16, 2023"
                  >
                    16
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 17, 2023"
                  >
                    17
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 18, 2023"
                  >
                    18
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 19, 2023"
                  >
                    19
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 20, 2023"
                  >
                    20
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 21, 2023"
                  >
                    21
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 22, 2023"
                  >
                    22
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 23, 2023"
                  >
                    23
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 24, 2023"
                  >
                    24
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 25, 2023"
                  >
                    25
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 26, 2023"
                  >
                    26
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 27, 2023"
                  >
                    27
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 28, 2023"
                  >
                    28
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__tile--now react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 29, 2023"
                  >
                    29
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 30, 2023"
                  >
                    30
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 31, 2023"
                  >
                    31
                  </abbr>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
